olive/engine/footprint.py

```diff
@@ -180,9 +173,18 @@ def create_footprints_by_model_ids(self, model_ids):
             nodes[model_id] = deepcopy(self.nodes[model_id])
         return Footprint(nodes=nodes, objective_dict=deepcopy(self.objective_dict))
 
-    def create_pareto_frontier(self):
-        self.mark_pareto_frontier()
-        rls = {k: v for k, v in self.nodes.items() if v.is_pareto_frontier}
+    def create_pareto_frontier(self, output_model_num, objective_dict):
```

Comment:
`output_model_num` can be `None` by default.
`objective_dict ` can be removed and use self.objective_dict instead.